Matrix.py

```diff
@@ -0,0 +1,181 @@
+"""Matrices calculator class.
+
+The matrices calculator can be used to add, sub, multiplication (by scalars or matrices) and scalar
+division. also you can compare matrices or using them as a dictionary key. the program only works
+with squared matrices.
+
+    Typical usage example:
+
+    x = Matrix(((1,2,3),(4,5,6),(6,7,8)))
+    y = Matrix.unity(3)
+    new_matrix = x * y
+    """
+
+
+class Matrix:
+    """A matrices calculator for basic math actions.
+
+        Attributes:
+            data(tuple of tuples): a Matrix- each line is a tuple in the major tuple.
+    """
+
+    def __init__(self, data):
+        if type(data) is not tuple:
+            raise ValueError("Invalid matrix- not a tuple")
+
+        for line in data:
+            if len(line) != len(data) or type(line) is not tuple:
+                raise ValueError("Invalid matrix- lines must be tuples and as long as the matrix")
+
+        self.data = data
+
+    def __iter__(self):
+        return iter(self.data)
+
+    def __get__(self, instance, owner):
+        return self.data
+
+    def __getitem__(self, item):
+        return self.data[item]
+
+    def __repr__(self):
+        return f"{self.__class__.__name__}({self.data})"
+
+    def __str__(self):
+        matrix_in_lines = ""
+        for i in range(len(self.data)):
+            matrix_in_lines += (str(self.data[i]) + "\n")
+        return matrix_in_lines
+
+    @staticmethod
+    def unity(size):
+        """ Unity method creates a matrix of zeros with a diagonal of ones.
+
+        Args:
+            size(int)- the size of the matrix and lines.
+
+        Returns:
+            A matrix full of 0 and 1's diagonal.
+
+        Exceptions:
+            ValueError: for non-integers.
+        """
+        if type(size) is int:
+            # major_list will append sub lists with zeros and one if the index= major index.
+            major_list = [[1 if i == index else 0 for i in range(size)] for index in range(size)]
+            # converts list of lists to type Matrix. lt are the inside lists
+            return __class__((tuple([tuple(lt) for lt in major_list])))
```

Comment:
don't use __class__.... please read the assignment again and follow what they say there